error: may not be used on enums without variants
 --> tests/ui/variant.rs:6:1
  |
6 | enum EnumWZV {} //~ ERROR may not be used on enums without variants
  | ^^^^^^^^^^^^^^^

error: may not be used on enums with discriminants
  --> tests/ui/variant.rs:10:9
   |
10 |     A = 2, //~ ERROR may not be used on enums with discriminants
   |         ^

error: may not be used on enums with variants with zero fields
  --> tests/ui/variant.rs:16:5
   |
16 |     A, //~ ERROR may not be used on enums with variants with zero fields
   |     ^

error: may not be used on enums with variants with zero fields
  --> tests/ui/variant.rs:22:5
   |
22 |     A(), //~ ERROR may not be used on enums with variants with zero fields
   |     ^^^

error: may not be used on enums with variants with multiple fields
  --> tests/ui/variant.rs:29:5
   |
29 |     B(A, B), //~ ERROR may not be used on enums with variants with multiple fields
   |     ^^^^^^^

error: may not be used on enums with variants with named fields
  --> tests/ui/variant.rs:34:5
   |
34 |     A { x: A }, //~ ERROR may not be used on enums with variants with named fields
   |     ^^^^^^^^^^
