error: may not be used on enums without variants
 --> $DIR/variant.rs:4:14
  |
4 | enum EnumWZV {} //~ ERROR may not be used on enums without variants
  |              ^^

error: may not be used on enums with discriminants
 --> $DIR/variant.rs:8:9
  |
8 |     A = 2, //~ ERROR may not be used on enums with discriminants
  |         ^

error: may not be used on enums with variants with zero fields
  --> $DIR/variant.rs:14:5
   |
14 |     A, //~ ERROR may not be used on enums with variants with zero fields
   |     ^

error: may not be used on enums with variants with zero fields
  --> $DIR/variant.rs:20:5
   |
20 |     A(), //~ ERROR may not be used on enums with variants with zero fields
   |     ^^^

error: may not be used on enums with variants with multiple fields
  --> $DIR/variant.rs:27:5
   |
27 |     B(A, B), //~ ERROR may not be used on enums with variants with multiple fields
   |     ^^^^^^^

error: may not be used on enums with variants with named fields
  --> $DIR/variant.rs:32:5
   |
32 |     A { x: A }, //~ ERROR may not be used on enums with variants with named fields
   |     ^^^^^^^^^^
